### PR TITLE
New Module: snow_record - Create/Delete/Update records in ServiceNow

### PR DESCRIPTION
**Pull request by garbled1**
##### SUMMARY
Create/Delete/Update records in ServiceNow (SNOW)

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/notification/snow_record.py

##### ANSIBLE VERSION

```
ansible 2.4.0 (snow_record 925dc0ce30) last updated 2017/08/09 01:22:18 (GMT +000)
  config file = None
  configured module search path = [u'/home/ec2-user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ec2-user/ansible/lib/ansible
  executable location = /home/ec2-user/ansible/bin/ansible
  python version = 2.7.12 (default, Sep  1 2016, 22:14:00) [GCC 4.8.3 20140911 (Red Hat 4.8.3-9)]

```

##### ADDITIONAL INFORMATION
This module allows you to lookup, edit, create and delete arbitrary records in ServiceNow.  https://www.servicenow.com

For example, one can update the CMDB entry for a machine with gathered ansible facts.  Or you could create an incident on a failed run of ansible, update work notes in a ticket when ansible runs against the host, etc etc.

```
- name: Create an incident
  snow_record:
    username: ansible_test
    password: my_password
    instance: dev99999
    state: present
    data:
      short_description: "This is a test incident opened by Ansible"
      severity: 3
      priority: 2
  register: new_incident

```

### CODE REVIEW

**Comment by garbled1** [comment]
I would really like to see this get into the base before 2.4 is frozen for contributed modules.  I have some more usage examples here:

[snow_record_tests](https://github.com/garbled1/ansible_modules/tree/master/roles/snow_record_test/tasks)

File: lib/ansible/modules/notification/snow_record.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+            - Table to query for records
+        required: false
+        default: incident
+    state:
+        description:
+            - If C(present) or C(updated) is supplied with a C(number)
+              argument, the module will attempt to update the record with
+              the supplied data.  If no such record exists, a new one will
+              be created.  C(absent) will delete a record.
+        choices: [ present, absent, updated ]
```
Comment (bcoca) [inline lib/ansible/modules/notification/snow_record.py]: updated should be implied in present ... its not a good pattern for state

Comment (garbled1) [reply lib/ansible/modules/notification/snow_record.py]: Fixed per your suggestion.  Thank you.

**Comment by garbled1** [comment]
Does this module need anything else?  Be glad to fixup anything it requires.

**Comment by bcoca** [comment]
rebuild_merge

**Pull request merged**